sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -36,29 +38,20 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
+        StringBuilder message = new StringBuilder();
         return Flux.fromIterable(credentials)
                    .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
-                       exceptions.add(t);
+                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
+                           throw logger.logExceptionAsError(new CredentialUnavailableException(
+                                   unavailableError + p.getClass().getSimpleName() + " authentication failed.",
+                                   t));
+                       }
+                       message.append(t.getMessage()).append(" "); 
                        return Mono.empty();
                    }), 1)
                    .next()
                    .switchIfEmpty(Mono.defer(() -> {
-
-                       StringBuilder message = new StringBuilder("Tried "
-                             + credentials.stream().map(c -> c.getClass().getSimpleName())
-                                   .collect(Collectors.joining(", "))
-                             + " but failed to acquire a token for any of them. Please verify the"
-                             + " environment for the credentials"
-                             + " and see more details in the causes below.");
-
-                       // Chain Exceptions.
-                       CredentialUnavailableException last = exceptions.get(exceptions.size() - 1);
-                       for (int z = exceptions.size() - 2; z >= 0; z--) {
-                           CredentialUnavailableException current = exceptions.get(z);
-                           last = new CredentialUnavailableException(current.getMessage(), last);
-                       }
-                       return Mono.error(new CredentialUnavailableException(message.toString(), last));
+                       return Mono.error(new CredentialUnavailableException(message.toString()));
```

Comment:
This code confuses me a bit, first we resume the error, potentially throwing or resuming with empty, then if it is the latter of empty we return an error. Could this be streamlined at all in the earlier code to bypass this operator? Such as using `onErrorMap` above.